main.js

```diff
@@ -46,15 +46,22 @@ function createNewIdea() {
 // May refactor to:
   // var newIdea = new Idea(titleInput.value, bodyInput.value)
   newIdea.saveToStorage();//this calls localStorage
-  list.push(newIdea);//pushes to array
+  // list.push(newIdea);//pushes to array
   displayCard()
   clearInputs()
 }
 
 function displayCard() {
   ideaSection.innerHTML = '';
+  //lines 56 - 61 below do bring back local storage on load, but everytime something is favorited it's duplicated
+  for(var i = 0; i < localStorage.length; i++){
+    var key = localStorage.key(i);
+    var storedCard = JSON.parse(localStorage.getItem(key));
+    var newStoredCard = new Idea(/*storedCard.id, */storedCard.title, storedCard.body, storedCard.star); //throwing the id in here jumbles the order of the card
+    list.push(newStoredCard);
+  }
   for(var i = 0; i < list.length; i++) {
-//we can definitely refactor by taking out a bunch of these vars
+//we can definitely refactor this to make it DRY code. It's very repetitive.
     var ideaTitle = list[i].title;
     var ideaBody = list[i].body;
     var ideaId = list[i].id;
```

Comment:
Proposed refactor of deleteCard: 
function deleteCard(id) {
  pullFromStorage()
  for (var i = 0; i < list.length; i++) {
    if (list[i].id === id) {
      localStorage.removeItem(`${id}`)
    }
  }
  displayCard();
}

--
proposed refactor of favoriteCard:
function favoriteCard(id) {
  pullFromStorage()
  for (var i = 0; i < list.length; i++) {
    if (id === list[i].id) {
      list[i].updateIdea();
      list[i].saveToStorage();
    }
    displayCard();
  }

-- proposed new pullFromStorage function:
function pullFromStorage() {
  list = [];
  for (var i = 0; i < localStorage.length; i++) {
    var key = localStorage.key(i);
    var storedCard = JSON.parse(localStorage.getItem(key));
    var instantiatedCard = new Idea(storedCard.title, storedCard.body, storedCard.star, storedCard.id);
    list.push(instantiatedCard)
  }
}